src/ucp/core/ucp_worker.c

```diff
@@ -679,6 +679,8 @@ ucp_worker_iface_error_handler(void *arg, uct_ep_h uct_ep, ucs_status_t status)
     if (ucp_worker_is_uct_ep_discarding(worker, uct_ep)) {
         ucs_debug("UCT EP %p is being discarded on UCP Worker %p",
                   uct_ep, worker);
+        uct_ep_pending_purge(uct_ep, ucp_ep_err_pending_purge,
```

Comment:
why needed?
UCT EP should be purged as a part of discarding procedure

Reply:
discard itself can be on pending

Reply:
> discard itself can be on pending

but why we need to remove discarding from the pending?

Reply:
it should be removed by discarding

Reply:
it does not remove, `ucp_ep_err_pending_purge` does `ucp_request_send_state_ff` which posts flush cancel again, that's the fix

Reply:
pls add comment to describe why it's needed